src/main/java/com/eu/habbo/habbohotel/users/HabboItem.java

```diff
@@ -556,4 +560,61 @@ public Rectangle getRectangle(int marginX, int marginY) {
                 this.getBaseItem().getLength() + (marginY * 2),
                 this.getRotation());
     }
+
+    public boolean isItemHiddenByAreaHider() {
```

Comment:
Seems like a very slow method to run for **every item** when someone enters the room. Every time someone enters a room, it runs this method for every item, which iterates over every item (to get the area hiders), decodes the json and checks if it should be hidden.